### PR TITLE
[FEAT] @Entry

### PR DESCRIPTION
**Pull request by MiaKoring**
### Summary
Adds the `@Entry` macro to streamline defining custom values for EnvironmentValues and AppStorageValues, replacing the need for developer written boilerplate of manual Key protocol implementation and accessor logic.

### Usage:
```swift
extension EnvironmentValues {
    @Entry
    var foo = Foo()
}

extension AppStorageValues {
    @Entry
    var foo = Foo()
}
```

### Changes
- New `@Entry` Macro: Automates key generation and storage accessors.
- AppStorage Update: Added `init(_: WritableKeyPath<AppStorageValues, Value>)` to support the new key-path based lookup.
- Internal Refactor: Migrated most SCUI EnvironmentValues to `@Entry`.
- Performance Optimization: Kept high-frequency values (e.g., `allowLayoutCaching`) as stored properties to avoid dictionary lookup overhead.
- Examples & Tests: Updated GreetingGeneratorExample and ControlsExample; added comprehensive unit tests.

### Notes
Due to `AppStorageValues` not being self-contained, it uses different accessors than `EnvironmentValues` and also has an additional value in its Key.

Stored and `@Entry` properties are fully interchangeable, so in case a certain value causes performance issues it can just be moved back to being stored.

### CODE REVIEW

File: Examples/Sources/GreetingGeneratorExample/GreetingGeneratorApp.swift [hunk 1]
```diff
@@ -68,8 +68,6 @@ struct LatestGreetingKey: EnvironmentKey {
 }
```
Comment (stackotter) [inline Examples/Sources/GreetingGeneratorExample/GreetingGeneratorApp.swift]: I believe that this can be removed

File: Sources/SwiftCrossUI/Environment/EnvironmentValues.swift [hunk 2]
```diff
@@ -44,89 +24,27 @@ public struct EnvironmentValues {
...
-    private var extraValues: [ObjectIdentifier: Any]
-
-    /// A mapping of window IDs to functions that open the corresponding windows.
-    var openWindowFunctionsByID: Box<[String: @MainActor () -> Void]>
+    private var values: [ObjectIdentifier: Any]
 
-    /// Backing storage for observable subscript
-    private var observableObjects: [ObjectIdentifier: any ObservableObject]
+    /// Whether user interaction is enabled. Set by ``View/disabled(_:)``.
+    public var isEnabled: Bool = true
```
Comment (stackotter) [inline Sources/SwiftCrossUI/Environment/EnvironmentValues.swift]: Why isn't this one an entry? It only changes when the `disabled` modifier is used right?

File: Sources/SwiftCrossUIMacrosPlugin/Utils/VariableExtension.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                    ] {
+                        return true
+                    }
+                default:
+                    break
+            }
+        }
+        return false
+    }
+}
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/Utils/VariableExtension.swift:26]: In the `CustomCodableMacro` from MacroToolkit's example plugin, we perform a similar check simply by doing `guard let attribute = variable.attributes.first(called: name) else { ... }`

File: Sources/SwiftCrossUI/State/StateImpl.swift [hunk 4]
```diff
@@ -1,4 +1,4 @@
-struct StateImpl<Storage: StateStorageProtocol> {
+public struct StateImpl<Storage: StateStorageProtocol> {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/StateImpl.swift]: Does this have to be made public? It's an internal implementation detail that I'd rather not expose if we don't have to. To fully support user-defined dynamic stateful properties we'll probably do something similar to SwiftUI.

File: Sources/SwiftCrossUI/State/StateImpl.swift [hunk 5]
```diff
@@ -58,7 +58,7 @@ struct StateImpl<Storage: StateStorageProtocol> {
     }
 }
 
-protocol StateStorageProtocol: AnyObject {
+public protocol StateStorageProtocol: AnyObject {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/StateImpl.swift]: Same as for StateImpl, does this have to be made public?

File: Sources/SwiftCrossUI/State/AppStorage.swift [hunk 6]
```diff
@@ -176,3 +203,21 @@ public protocol AppStorageKey<Value> {
     /// The default value for the key.
     static var defaultValue: Value { get }
 }
+
+public struct AppStorageValues {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage.swift]: Move this to a separate file given that it's public and not completely trivial. We should probably do that with AppStorageKey as well, but that's a previous PR's fault.

File: Sources/SwiftCrossUI/State/AppStorage.swift [hunk 7]
```diff
@@ -164,6 +136,61 @@ extension AppStorage {
     public init<Key: AppStorageKey<Value>>(_: Key.Type) {
         self.init(wrappedValue: Key.defaultValue, Key.name)
     }
+
+    public init(_ keyPath: WritableKeyPath<AppStorageValues, Value>) {
+        implementation = StateImpl(initialStorage: Storage(mode: .path(keyPath)))
+    }
+}
+
+// MARK: - AppStorageProviderExtension
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage.swift]: Remove this mark

File: Sources/SwiftCrossUI/State/AppStorage.swift [hunk 8]
```diff
@@ -164,6 +136,61 @@ extension AppStorage {
...
+extension AppStorageProvider {
+    public func getValue<T: Codable>(key: String, defaultValue: T) -> T {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage.swift]: Require T to be Sendable to silence the warnings.

File: Sources/SwiftCrossUI/State/AppStorage.swift [hunk 9]
```diff
@@ -164,6 +136,61 @@ extension AppStorage {
...
+                    ]
+                )
+                return defaultValue
+            }
+
+            return cachedValue
+        }
+    }
+
+    public func setValue<T: Codable>(key: String, newValue: T) {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage.swift]: Require T to be Sendable to silence the warnings.

File: Sources/SwiftCrossUI/State/AppStorage.swift [hunk 10]
```diff
@@ -176,3 +203,21 @@ public protocol AppStorageKey<Value> {
...
+
+public struct AppStorageValues {
+    private let __provider: AppStorageProvider?
+
+    /// Only to be used by AppStorage
+    internal init(__provider: AppStorageProvider?) {
+        self.__provider = __provider
+    }
+
+    public func __getValue<T: Codable>(_ key: AppStorageKey<T>.Type) -> T {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage.swift]: Fix these warnings with `any`

File: Sources/SwiftCrossUI/State/AppStorage.swift [hunk 11]
```diff
@@ -176,3 +203,21 @@ public protocol AppStorageKey<Value> {
...
+        guard let __provider else { return key.defaultValue }
+        return __provider.getValue(key: key.name, defaultValue: key.defaultValue)
+    }
+
+    public func __setValue<T: Codable>(_ key: AppStorageKey<T>.Type, newValue: T) {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage.swift]: Add `any` to fix the warnings

File: Sources/SwiftCrossUI/State/AppStorage/AppStorage.swift [hunk 12]
```diff
@@ -164,6 +136,61 @@ extension AppStorage {
...
         self.init(wrappedValue: Key.defaultValue, Key.name)
     }
+
+    public init(_ keyPath: WritableKeyPath<AppStorageValues, Value>) {
+        implementation = StateImpl(initialStorage: Storage(mode: .path(keyPath)))
+    }
+}
+
+// MARK: - AppStorageProviderExtension
+extension AppStorageProvider {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage/AppStorage.swift]: Can this be in the AppStorageProvider file?

File: Tests/SwiftCrossUITests/EntryMacroTests.swift [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+            """,
+            expandedSource: """
+            extension EnvironmentValues {
+                var test: UInt64
+            
+                private struct __Key_test: SwiftCrossUI.EnvironmentKey {
+            
+                }
+            }
+            """,
```
Comment (stackotter) [inline Tests/SwiftCrossUITests/EntryMacroTests.swift:87]: Can we omit the `expandedSource` here given that we don't actually care what it is?

File: Tests/SwiftCrossUITests/Utils/TestFailureSpecExtension.swift [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import SwiftSyntaxMacrosGenericTestSupport
+import Testing
+
+extension TestFailureSpec {
+    public var issueComment: Comment {
+        "\(message) \nat: \(location.filePath) \(location.line):\(location.column)"
```
Comment (stackotter) [inline Tests/SwiftCrossUITests/Utils/TestFailureSpecExtension.swift:6]: Might be best to keep the comment to a single line; e.g. `\(message) (at: ...)`

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+                    private struct __Key_\(identifier): SwiftCrossUI.\(keyName.rawValue)Key {
+                        \(defaultValueDeclaration)\(nameDeclaration)
+                    } 
+                    """)
+        ]
+
+        return []
+    }
+
+    enum KeyName: String {
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: I don't really understand the name of this enum. `EnclosingType` might be better? (because these aren't key names)

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+    public static func expansion(
+        of node: SwiftSyntax.AttributeSyntax,
+        providingAccessorsOf declaration: some SwiftSyntax.DeclSyntaxProtocol,
+        in context: some SwiftSyntaxMacros.MacroExpansionContext
+    ) throws -> [SwiftSyntax.AccessorDeclSyntax] {
+        guard
+            let extensionDecl = context.lexicalContext.compactMap({ syntax in
+                syntax.as(ExtensionDeclSyntax.self)
+            }).first
+        else { return [] }
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: We want this macro to fail if the extension isn't the direct parent of the `@Entry`-annotated property, so we should require that here.

E.g. you current implementation might accept this and then fail in a confusing way later on during compilation;

```swift
extension EnvironmentValues {
    struct Asdf {
        @Entry var name: String?
    }

    // or

    var asdf: String? {
        @Entry var hello: String?
    }
}
```

Comment (stackotter) [reply Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: We should also throw a `MacroError` here with a description like "`@Entry`-annotated properties must be direct children of an EnvironmentValues or AppStorageValues extension."

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+        let extendedType = extensionDecl.extendedType.trimmedDescription
+
+        guard let keyName = KeyName(rawValue: extendedType) else { return [] }
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: Throw an error

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+
+        guard
+            let variable = Decl(declaration).asVariable,
+            let patternBinding = destructureSingle(variable.bindings),
+            let identifier = patternBinding.identifier
+        else { return [] }
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: Throw an error

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+
+        if patternBinding.initialValue == nil,
+            patternBinding.type?.isOptional != true
+        {
+            throw MacroError("@Entry requires an initial value to be set.")
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: `@Entry requires an initial value for non-optional properties`

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+        of node: SwiftSyntax.AttributeSyntax,
+        providingPeersOf declaration: some SwiftSyntax.DeclSyntaxProtocol,
+        in context: some SwiftSyntaxMacros.MacroExpansionContext
+    ) throws -> [SwiftSyntax.DeclSyntax] {
+        // Information about extension context
+        guard
+            let extensionDecl = context.lexicalContext.compactMap({ syntax in
+                syntax.as(ExtensionDeclSyntax.self)
+            }).first
+        else { return [] }
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: Same as comment at top of previous method. We want direct parent to be an extension declaration.

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+        let extendedType = extensionDecl.extendedType.trimmedDescription
+
+        guard let keyName = KeyName(rawValue: extendedType) else { return [] }
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: Throw error (same for other silent early exits below)

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+
+        // Optional types get nil as default value if no initial value is provided
+        var defaultValueDeclaration = ""
+        if patternBinding.initialValue == nil,
+            patternBinding.type?.isOptional == true
+        {
+            defaultValueDeclaration = "static let defaultValue\(typeDeclaration) = nil"
+        } else if let initialValue = patternBinding.initialValue?._syntax.trimmedDescription {
+            defaultValueDeclaration = "static let defaultValue\(typeDeclaration) = \(initialValue)"
+        }  // No else case here, as it already throws at the accessor macro
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: It'd be good to throw here so that the `defaultValueDeclaration` variable doesn't have to have the theoretically never-used `""` initial value (and can be a `let`). We also can't assume the ordering of these macro expansions, so this one might get expanded first and then cause type checking errors before the other one gets expanded.

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+            case .environment:
+                nameDeclaration = ""
+            case .appStorage:
+                nameDeclaration = "\nstatic let name = \"\(identifier)\""
+        }
+
+        return [
+            DeclSyntax(
+                stringLiteral: """
+                    private struct __Key_\(identifier): SwiftCrossUI.\(keyName.rawValue)Key {
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: Add a `keyName` computed property to `KeyName` (which I've suggested renaming to `EnclosingType`, and return `"SwiftCrossUI.\(self.rawValue)Key"` from there to centralize things nicely.

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+                        \(defaultValueDeclaration)\(nameDeclaration)
+                    } 
+                    """)
+        ]
+
+        return []
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: Remove unused return

File: Sources/SwiftCrossUI/Environment/EnvironmentValues.swift [hunk 25]
```diff
@@ -328,10 +207,123 @@ public struct EnvironmentValues {
...
 /// A key that can be used to extend the environment with new properties.
 public protocol EnvironmentKey<Value> {
     /// The type of value the key can hold.
     associatedtype Value
     /// The default value for the key.
     static var defaultValue: Value { get }
 }
+
+@attached(accessor) @attached(peer, names: prefixed(__Key_))
+public macro Entry() =
```
Comment (stackotter) [inline Sources/SwiftCrossUI/Environment/EnvironmentValues.swift]: Please move this macro declaration to a separate file as well (Entry.swift), especially given that it's not just an EnvironmentValues thing. And add a documentation comment with example usage code for both EnvironmentValues and AppStorageValues.

File: Sources/SwiftCrossUI/Environment/EnvironmentValues.swift (hunk 2 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUI/Environment/EnvironmentValues.swift]: I had concerns about the hashing overhead, as its one of the more often accessed properties, I can change it and we see if it becomes a bottleneck at some point?

File: Sources/SwiftCrossUI/State/StateImpl.swift (hunk 4 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUI/State/StateImpl.swift]: I just thought it may be useful for custom implementations, but as it isn't the final idea I will make it internal again

File: Tests/SwiftCrossUITests/Utils/TestFailureSpecExtension.swift (hunk 14 above)
Comment (MiaKoring) [reply Tests/SwiftCrossUITests/Utils/TestFailureSpecExtension.swift:6]: the message is really long with code and diff, I originally had it in the same line, but the message would usually be a long multiline string, so I felt like another line was better. you can try by removing a character in the expanded source, making it fail

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift (hunk 15 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: You are right, that is better. The thought was that of xyzKey xyz would be the name (Environment would be the name, appended by Key becoming EnvironmentKey)

File: Sources/SwiftCrossUI/Environment/EnvironmentValues.swift (hunk 2 above)
Comment (stackotter) [reply Sources/SwiftCrossUI/Environment/EnvironmentValues.swift]: Lots of other properties (such as the font) are accessed a similar amount, so yeah we should probably leave this particular optimisation for future performance work

**Comment by stackotter** [comment]
Oh and UIKitBackend has a custom environment value at around `Sources/UIKitBackend/KeyboardToolbar.swift:300`

File: Sources/SwiftCrossUIMacrosPlugin/Utils/VariableExtension.swift (hunk 3 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUIMacrosPlugin/Utils/VariableExtension.swift:26]:
```
guard let targetAttribute = attributes.first(called: name) else { return false }
return targetAttribute.asMacroAttribute != nil
```
doesn’t work. I’m not entirely sure WHY it doesn’t. But the code I stole from https://github.com/swiftlang/swift/blob/main/lib/Macros/Sources/ObservationMacros/Extensions.swift does.
(In my next push I’ll have attribution, I forgot about the Apache V 2 License)

File: Sources/SwiftCrossUI/State/AppStorage/AppStorage.swift (hunk 12 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUI/State/AppStorage/AppStorage.swift]: It cannot, unless 
```swift
private let appStorageCache: Mutex<[String: any Codable & Sendable]> = Mutex([:])
```
isn’t private anymore

File: Tests/SwiftCrossUITests/EntryMacroTests.swift (hunk 13 above)
Comment (MiaKoring) [reply Tests/SwiftCrossUITests/EntryMacroTests.swift:87]: We cannot, expandedSource is a non-optional parameter

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift (hunk 21 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: I’m not convinced it makes sense to throw the same error twice. 
We would need to specify it 2 times in the tests for the error and the user would also get the same message twice one the same macro application.

File: Acknowledgements/swift/LICENSE.txt [hunk 26]
```diff
@@ -0,0 +1,211 @@
...
+                        http://www.apache.org/licenses/
+
+    TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+
+    1. Definitions.
+
+      "License" shall mean the terms and conditions for use, reproduction,
+      and distribution as defined by Sections 1 through 9 of this document.
+
+      "Licensor" shall mean the copyright owner or entity authorized by
```
Comment (stackotter) [inline Acknowledgements/swift/LICENSE.txt:12]: Did you mean for this to be in this PR?

File: Sources/SwiftCrossUI/Environment/EnvironmentValues.swift [hunk 27]
```diff
@@ -290,18 +287,46 @@ extension EnvironmentValues {
     /// The menu ordering to use.
     @Entry public var menuOrder: MenuOrder = .automatic
 
+    @Entry private var openWindowFunctionsByIDStore = UncheckedSendable(
+        wrappedValue: Box<[String: @MainActor () -> Void]>([:]))
```
Comment (stackotter) [inline Sources/SwiftCrossUI/Environment/EnvironmentValues.swift:293]: Is this pattern of a private `blahStore` and a public computed `blah` property to address compiler warnings? If so, please document each of these private `blahStore` properties to make their purpose clear.

File: Sources/SwiftCrossUIMacrosPlugin/Utils/VariableExtension.swift [hunk 28]
```diff
@@ -4,6 +4,10 @@ import SwiftSyntaxMacroExpansion
 import SwiftSyntaxMacros
 
 extension Variable {
+    // Portions of this code are derived from the Swift Open Source Project.
+    // Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+    // Licensed under Apache License v2.0 with Runtime Library Exception
+    // Source https://github.com/swiftlang/swift/blob/2e8977f
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/Utils/VariableExtension.swift]: Link to the actual source of this algorithm

File: Sources/SwiftCrossUI/State/AppStorage/AppStorageValues.swift [hunk 29]
```diff
@@ -0,0 +1,17 @@
+public struct AppStorageValues {
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage/AppStorageValues.swift:2]: Document this type. Something like `A container for typed application-defined app storage values`.

File: Sources/SwiftCrossUI/State/AppStorage/AppStorageKey.swift [hunk 30]
```diff
@@ -0,0 +1,10 @@
+/// A type safe key for ``AppStorage`` properties, similar in spirit
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage/AppStorageKey.swift]: Should probably say AppStorageValues instead of AppStorage given the properties are on AppStorageValues. You could then add a sentence to explain that they are accessed using the AppStorage property wrapper.

File: Sources/SwiftCrossUI/State/AppStorage/AppStorageValues.swift [hunk 31]
```diff
@@ -0,0 +1,17 @@
...
+    private let __provider: AppStorageProvider?
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/AppStorage/AppStorageValues.swift]: This only just came to mind, but why the double leading underscore instead of a single leading underscore?

I've also realised that given that `__provider` is private, user-defined properties can't clash with it anyway, so it probably doesn't have to be underscored at all.

Also, from a quick local test, Swift handles properties that have the same name as a method just fine, so the methods can probably have their double underscores removed as well.

File: Sources/SwiftCrossUI/State/AppStorage/AppStorage.swift (hunk 12 above)
Comment (stackotter) [reply Sources/SwiftCrossUI/State/AppStorage/AppStorage.swift]: I'd be ok with removing the `private` on that cache to allow this extension to move into the AppStorageProvider file

File: Sources/SwiftCrossUI/State/Entry.swift [hunk 32]
```diff
@@ -0,0 +1,23 @@
+/// Creates an environment values, or appstorage values entry.
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/Entry.swift]: Link these to EnvironmentValues and AppStorageValues properly

File: Sources/SwiftCrossUI/State/Entry.swift [hunk 33]
```diff
@@ -0,0 +1,23 @@
...
+///
+/// Create EnvironmentValues entries by extending the EnvironmentValues structure with new properties and attaching the @Entry macro to the variable declarations:
```
Comment (stackotter) [inline Sources/SwiftCrossUI/State/Entry.swift]:
```
You can create ``EnvironmentValues`` entries by extending the ``EnvironmentValues`` struct with `@Entry`-annotated properties:
```

Use that same wording for the appstorage as well (replacing EnvironmentValues with AppStorageValues)

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 34]
```diff
@@ -63,22 +51,68 @@ public struct EntryMacro: AccessorMacro, PeerMacro {
...
+        defaultValueDeclaration: String
+    ) {
+        // Verify extension context
+        guard
+            let extensionDecl = context.lexicalContext.first?.as(ExtensionDeclSyntax.self),
+            let enclosingValueType = EnclosingType(
+                rawValue: extensionDecl.extendedType.trimmedDescription)
+        else {
+            throw MacroError(
+                "@Entry-annotated properties must be direct children of an EnvironmentValues or AppStorageValues extension."
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: of EnvironmentValues or AppStorageValues extensions

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 35]
```diff
@@ -63,22 +51,68 @@ public struct EntryMacro: AccessorMacro, PeerMacro {
...
+                nameDeclaration = ""
+            case .appStorage:
+                nameDeclaration = "\nstatic let name = \"\(identifier)\""
+        }
 
-        // Information about variable
+        return [
+            DeclSyntax(
+                stringLiteral: """
+                    private struct __Key_\(identifier): \(enclosingType.keyName) {
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: We should probably surround our key struct identifier with backticks, because from Swift 6.2 developers can have spaces in variable identifiers (when surrounded by backticks).

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 36]
```diff
@@ -63,22 +51,68 @@ public struct EntryMacro: AccessorMacro, PeerMacro {
...
+        }
 
-        guard let keyName = KeyName(rawValue: extendedType) else { return [] }
+        // AppStorage has got a special requirement to know the key name as string
+        let nameDeclaration: String
+        switch enclosingType {
+            case .environment:
+                nameDeclaration = ""
+            case .appStorage:
+                nameDeclaration = "\nstatic let name = \"\(identifier)\""
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: We should escape `"` characters in `identifier`, because with [raw identifiers](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0451-escaped-identifiers.md#permitted-characters) developers can have double quotes in variable names.

`identifier.replacingOccurrences(of: "\"", with: "\\\"")`

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift [hunk 37]
```diff
@@ -87,38 +121,25 @@ public struct EntryMacro: AccessorMacro, PeerMacro {
             typeDeclaration = ""
         }
 
-        // Optional types get nil as default value if no initial value is provided
+        // Verify defaultValue
         var defaultValueDeclaration = ""
```
Comment (stackotter) [inline Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: I think that this can be a let without a default value

File: Tests/SwiftCrossUITests/EntryMacroTests.swift [hunk 38]
```diff
@@ -194,5 +190,63 @@ struct EntryMacroTests {
...
+                    line: 2,
+                    column: 5
+                )
+            ],
+            macroSpecs: testMacros,
+            failureHandler: { spec in
+                Issue.record(spec.issueComment)
+            }
+        )
+    }
```
Comment (stackotter) [inline Tests/SwiftCrossUITests/EntryMacroTests.swift:250]: Add a test for backticked variable names, because I feel like they could be an interesting edge case.

**Review by stackotter — CHANGES_REQUESTED** [review]

File: Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift (hunk 35 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: I'm pretty sure those wouldn't be picked up with the current SwiftSyntax version (they didn't in my test function parsing)

can I update it?

Comment (stackotter) [reply Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: We should leave it at the current version for now because it has to be the same as Swift Bundler. I realise that it doesn’t support actually parsing raw identifiers, but when we update our Swift Syntax version there won’t be any API breaking change to remind us of this issue, and that issue will silently get introduced. So if we handle it now (cause we can without the new API, using string replacement), it’s future proof, if I’m understanding things correctly

Comment (MiaKoring) [reply Sources/SwiftCrossUIMacrosPlugin/EntryMacro.swift]: It should be, we just can't add a test, which sucks a little.

File: Acknowledgements/swift/LICENSE.txt (hunk 26 above)
Comment (MiaKoring) [reply Acknowledgements/swift/LICENSE.txt:12]: Yes, I used code from the swift repo, which is Apache licensed and afaik a copy of the license is an Apache license requirement.

Comment (stackotter) [reply Acknowledgements/swift/LICENSE.txt:12]: Oh, sorry, I meant to remove this comment when I saw the source of the attribution 🤦‍♂️

File: Sources/SwiftCrossUI/State/Entry.swift (hunk 33 above)
Comment (MiaKoring) [reply Sources/SwiftCrossUI/State/Entry.swift]: done, I wonder why the Apple documentation doesn’t link to the struct declarations. (I took the comment from there)

File: Tests/SwiftCrossUITests/EntryMacroTests.swift (hunk 38 above)
Comment (MiaKoring) [reply Tests/SwiftCrossUITests/EntryMacroTests.swift:250]: Just adding the comment here again, so its not forgotten, won’t work without a swift-syntax version bump to 602.x.x

File: Sources/SwiftCrossUI/State/Entry.swift (hunk 33 above)
Comment (stackotter) [reply Sources/SwiftCrossUI/State/Entry.swift]: Interesting...
